hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java

```diff
@@ -277,6 +277,50 @@ public JobConfig addJar(@Nonnull String path) {
         }
     }
 
+    /**
+     * Adds the ZIP file identified by the supplied URL to the list of archives
+     * that will be a part of the job's classpath while it's executing in the
+     * Jet cluster.
```

Comment:
From this description it wasn't clear to me what's the difference between jar and zip. We load all `jar` files inside the zip file, which was surprising to me.

When you are specifying classpath for `java` and you specify a zip file, afaik, it will be treated just like a jar file - java ignores the extension. Actually you can't specify zip-with-jars file using `java -cp`.

Also what's surprising that it doesn't matter whether you use `addJar` or `addZip`, only the actual file extension matters. If you add a file ending with `zip` using the `addJar` method because it has methods directly in it, no error will be reported and no classes will be loaded. Also, the extension checking is case sensitive even on windows. It would be better to handle the file according to which method was used and ignore the actual file extension. And I'd rename this method to `addJarsInZip`.

Reply:
I propose closing this PR and merging this work with my Python work, there's a lot of overlap and we have to solve these two use cases so they don't collide.